cpp/src/gandiva/gdv_function_stubs_test.cc

```diff
@@ -160,4 +160,13 @@ TEST(TestGdvFnStubs, TestCastFloat8) {
   ctx.Reset();
 }
 
+TEST(TestGdvFnStubs, TestIsDate) {
+  EXPECT_EQ(gdv_fn_is_date_utf8("2020-01-01", 10), true);
+  EXPECT_EQ(gdv_fn_is_date_utf8("1524-12-31", 10), true);
+  EXPECT_EQ(gdv_fn_is_date_utf8("0001-01-01", 10), true);
+  EXPECT_EQ(gdv_fn_is_date_utf8("2020-01-01 12:00:00", 19), false);
```

Comment:
this should be a valid date i think

Reply:
Yes, you are right, the ISDATE function should return true since it is castable to a data YYYY-MM-DD format.
Corrected the behavior of the function